src/test/java/mate/academy/dao/RoleDaoTest.java

```diff
@@ -0,0 +1,75 @@
+package mate.academy.dao;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Optional;
+import mate.academy.dao.impl.RoleDaoImpl;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.Role;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RoleDaoTest extends AbstractTest {
+    private static final String USER = "USER";
+    private static final String NOT_USER = "NOT_USER";
+    private RoleDao roleDao;
+    private Role userRole;
+    private Role actual;
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[] {Role.class};
+    }
+
+    @BeforeEach
+    void setUp() {
+        roleDao = new RoleDaoImpl(getSessionFactory());
+        userRole = new Role(Role.RoleName.USER);
+        actual = roleDao.save(userRole);
+    }
+
+    @Test
+    void save_Ok() {
+        assertNotNull(actual);
+        assertEquals(USER, actual.getRoleName().name());
```

Comment:
if we explicitly test save logic it's better to call it here to see full cycle, possibly with some different name to be sure that this particular save operation works as intended